tools/clang/test/CodeGenSPIRV/intrinsics.asuint.hlsl

```diff
@@ -76,11 +76,34 @@ void main() {
     double value;
     uint lowbits;
     uint highbits;
-// CHECK-NEXT:      [[value:%[0-9]+]] = OpLoad %double %value
-// CHECK-NEXT:  [[resultVec:%[0-9]+]] = OpBitcast %v2uint [[value]]
-// CHECK-NEXT: [[resultVec0:%[0-9]+]] = OpCompositeExtract %uint [[resultVec]] 0
+// CHECK-NEXT:  [[resultArr:%[0-9]+]] = OpBitcast %_ptr_Function__arr_uint_uint_2 %value
```

Comment:
You can't do this. A bitcast from one pointer type to another is not allowed in the logical addressing mode.

See https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_universal_validation_rules. In particular, the list of valid opcodes for which a pointer can be an operand.

Reply:
Hm, I think in that case I can maybe do component-wise bitcasts for each member of the vector.

Is there a reason the validator doesn't catch this?